### ISSUE TITLE
Not working with Webpack 4.19.1

### ISSUE THREAD

**Issue by ehmadzubair** [django-webpack/django-webpack-loader#183]
The following outputs a blank page

Webpack: 4.19.1

**Index.html Template**
```{% load render_bundle from webpack_loader %}
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title>React Django</title>
  </head>
  <body>
    <div id="root">
    </div>
      {% render_bundle 'main' %}
  </body>
</html>
```

**Rendered Index**
```<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title>React Django</title>
  </head>
  <body>
    <div id="root">
    </div>
      <script type="text/javascript" src="http://localhost:3000/static/js/main.chunk.js" ></script>
  </body>
</html>
```

**Expected output**
```<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title>React Django</title>
  </head>
  <body>
    <div id="root">
    </div>
      <script src="http://localhost:3000/static/js/bundle.js"></script>
     <script src="http://localhost:3000/static/js/1.chunk.js"></script>
     <script src="http://localhost:3000/static/js/main.chunk.js"></script>
  </body>
</html>
```

**Comment by jaisonjjames** [comment]
+1 Subscribed

**Comment by jaisonjjames** [comment]
> **TL:DR;** Resolved the issue
> 
> This is what my webpack-stats.dev.json looked like
> 
> ```
> {
>   "status": "done",
>   "publicPath": "http://localhost:3000/",
>   "chunks": {
>     "main": [
>       {
>         "name": "static/js/main.chunk.js",
>         "publicPath": "http://localhost:3000/static/js/main.chunk.js",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/main.chunk.js"
>       },
>       {
>         "name": "static/js/main.chunk.js.map",
>         "publicPath": "http://localhost:3000/static/js/main.chunk.js.map",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/main.chunk.js.map"
>       }
>     ],
>     "undefined": [
>       {
>         "name": "static/js/0.chunk.js",
>         "publicPath": "http://localhost:3000/static/js/0.chunk.js",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/0.chunk.js"
>       },
>       {
>         "name": "static/js/0.chunk.js.map",
>         "publicPath": "http://localhost:3000/static/js/0.chunk.js.map",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/0.chunk.js.map"
>       }
>     ],
>     "runtime~main": [
>       {
>         "name": "static/js/bundle.js",
>         "publicPath": "http://localhost:3000/static/js/bundle.js",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/bundle.js"
>       },
>       {
>         "name": "static/js/bundle.js.map",
>         "publicPath": "http://localhost:3000/static/js/bundle.js.map",
>         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/bundle.js.map"
>       }
>     ]
>   }
> }
> ```
> So, instead of just doing
> 
> ```
> {% render_bundle "main" %}
> ```
> I changed my `index.html` to be like so:
> 
> ```
> {% load render_bundle from webpack_loader %}
> <!DOCTYPE html>
> <html>
>   <head>
>     <meta charset="UTF-8" />
>     <meta name="viewport" content="width=device-width" />
>     <title>React Django</title>
>   </head>
>   <body>
>     <div id="root">
>     </div>
>       {% render_bundle "main" %}
>       {% render_bundle "undefined" %}
>       {% render_bundle "runtime~main" %}
>   </body>
> </html>
> ```

Oh great, Thanks a lot!, its worked for me.

**Comment by ajiejot** [comment]
> > **TL:DR;** Resolved the issue
> > This is what my webpack-stats.dev.json looked like
> > ```
> > {
> >   "status": "done",
> >   "publicPath": "http://localhost:3000/",
> >   "chunks": {
> >     "main": [
> >       {
> >         "name": "static/js/main.chunk.js",
> >         "publicPath": "http://localhost:3000/static/js/main.chunk.js",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/main.chunk.js"
> >       },
> >       {
> >         "name": "static/js/main.chunk.js.map",
> >         "publicPath": "http://localhost:3000/static/js/main.chunk.js.map",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/main.chunk.js.map"
> >       }
> >     ],
> >     "undefined": [
> >       {
> >         "name": "static/js/0.chunk.js",
> >         "publicPath": "http://localhost:3000/static/js/0.chunk.js",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/0.chunk.js"
> >       },
> >       {
> >         "name": "static/js/0.chunk.js.map",
> >         "publicPath": "http://localhost:3000/static/js/0.chunk.js.map",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/0.chunk.js.map"
> >       }
> >     ],
> >     "runtime~main": [
> >       {
> >         "name": "static/js/bundle.js",
> >         "publicPath": "http://localhost:3000/static/js/bundle.js",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/bundle.js"
> >       },
> >       {
> >         "name": "static/js/bundle.js.map",
> >         "publicPath": "http://localhost:3000/static/js/bundle.js.map",
> >         "path": "/Users/ehmadzubair/Documents/Self/Mojolo/django-app/mojolo/frontend/dist/static/js/bundle.js.map"
> >       }
> >     ]
> >   }
> > }
> > ```
> > So, instead of just doing
> > ```
> > {% render_bundle "main" %}
> > ```
> > I changed my `index.html` to be like so:
> > ```
> > {% load render_bundle from webpack_loader %}
> > <!DOCTYPE html>
> > <html>
> >   <head>
> >     <meta charset="UTF-8" />
> >     <meta name="viewport" content="width=device-width" />
> >     <title>React Django</title>
> >   </head>
> >   <body>
> >     <div id="root">
> >     </div>
> >       {% render_bundle "main" %}
> >       {% render_bundle "undefined" %}
> >       {% render_bundle "runtime~main" %}
> >   </body>
> > </html>
> > ```
> 
> Oh great, Thanks a lot!, its worked for me.

do you have your repository? I still cant figure how to this, I am using webpack version 1

**Comment by shaton** [comment]
I had the same issue, and I applied the same solution, but now I was wondering if it is the right way to fix it or is there another better way. Or maybe we had something wrong in the configuration.

**Comment by this-is-mike** [comment]
the 'undefined' in the {% render_bundle 'undefined' %} tag is because it's missing a name in the webpack.config.js.
Look for: 
`splitChunks: {
        chunks: "all",
        name: "false"
      },`
and replace "false" with vendor.

**Comment by trthhrtz** [comment]
> the 'undefined' in the {% render_bundle 'undefined' %} tag is because it's missing a name in the webpack.config.js.
> Look for:
> `splitChunks: { chunks: "all", name: "false" },`
> and replace "false" with vendor.

it did not help.
      ```
splitChunks: {
        chunks: 'all',
        name: 'test',
      },


{% render_bundle "main" %}
{% render_bundle "test" %}
```
gives 

**Cannot resolve bundle test.**


      ```
splitChunks: {
        chunks: 'all',
        name: 'main',
      },


{% render_bundle "main" %}
```
Doest not work


@this-is-mike

**Comment by caseykey** [comment]
> > the 'undefined' in the {% render_bundle 'undefined' %} tag is because it's missing a name in the webpack.config.js.
> > Look for:
> > `splitChunks: { chunks: "all", name: "false" },`
> > and replace "false" with vendor.
> 
> it did not help.
> 
> ```
> splitChunks: {
>         chunks: 'all',
>         name: 'test',
>       },
> 
> 
> {% render_bundle "main" %}
> {% render_bundle "test" %}
> ```
> 
> gives
> 
> **Cannot resolve bundle test.**
> 
> ```
> splitChunks: {
>         chunks: 'all',
>         name: 'main',
>       },
> 
> 
> {% render_bundle "main" %}
> ```
> 
> Doest not work (blank page)
> 
> @this-is-mike

Did this issue become resolved? 

Perhaps trying this will work.

Edit the webpack.config.dev.js:

```
module.exports = {
    // ...
    optimization: {

        // ...
        // Automatically split vendor and commons
        // ...
        splitChunks: {
          chunks: 'async', // from 'all'
          name: true, // from 'false'
        },

        // ...

        runtimeChunk: false //from 'true'

    }
// ...
}
```

**Comment by rvlb** [comment]
We've merged #232 which brings compatibility with `webpack-bundle-tracker@1.0.0-alpha.1` (compatible with Webpack >=4). Release will follow soon.
